src/cache_memoize/__init__.py

```diff
@@ -117,7 +120,15 @@ def inner(*args, **kwargs):
             else:
                 result = cache.get(cache_key, MARKER)
             if result is MARKER:
-                result = func(*args, **kwargs)
+
+                try:
+                    result = func(*args, **kwargs)
+                except Exception as exception:
+                    if exception.__class__ in (cache_exceptions or []):
```

Comment:
This implementation does not cover exception inheritance. For example, if I used:
```python
    @cache_memoize(1000, cache_exceptions=[Exception])
    def run_calculations(parameter):
        # ...
        raise InvalidParameter
```
then the exception would not be cached. While the example may be extreme (we shouldn't be just catchin `Exception` most of the time), the general expectation from users might be one of `isinstance` instead of direct class match.

Reply:
Good point. It's irrelevant for our use-case, so I did not think of it, but that behaviour is more in line with what I would think of as a 'general' use-case.